### PR TITLE
Add tidb-ctl 

### PR DESCRIPTION
**Pull request by LiangShang**
add tidb-ctl and two commands:
- table showID
- table showStore

### CODE REVIEW

File: cmd/tidb-ctl/main.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/chzyer/readline"
+
```
Comment (disksing) [inline cmd/tidb-ctl/main.go]: please remove the empty line.

File: cmd/tidb-ctl/main.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	flag "github.com/spf13/pflag"
+
```
Comment (disksing) [inline cmd/tidb-ctl/main.go]: please remove the empty line.

File: tidbctl/command/core.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package command
+
+import (
+	"encoding/json"
+	"errors"
```
Comment (disksing) [inline tidbctl/command/core.go]: We use "github.com/juju/errors" instead of built-in errors package.

File: tidbctl/command/core.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
+package command
```
Comment (disksing) [inline tidbctl/command/core.go]: please add liscense header.

File: tidbctl/command/table_command.go [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+	"github.com/spf13/cobra"
+)
+
+// NewTableCommand is the entrance of table command
+func NewTableCommand() *cobra.Command {
+	m := &cobra.Command{
+		Use:   "table <sub_command> <db_name> <table_name>",
+		Short: "show the table information given the table name",
+		// Run:   showTableIDCommandFunc,
```
Comment (disksing) [inline tidbctl/command/table_command.go]: remove this line.

File: tidbctl/ctl.go [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tidbctl
+
+import (
+	"fmt"
+
```
Comment (disksing) [inline tidbctl/ctl.go]: remove this line.

File: tidbctl/ctl.go [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/juju/errors"
+
```
Comment (disksing) [inline tidbctl/ctl.go]: remove this line.

File: tidbctl/ctl.go [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+	"github.com/pingcap/tidb/tidbctl/command"
+	"github.com/spf13/cobra"
+)
+
+var (
+	errInvalidAddr = errors.New("Invalid TiDB address, Cannot get connect to it")
```
Comment (disksing) [inline tidbctl/ctl.go]: Cannot -> cannot

**Comment by tiancaiamao** [comment]
Please do one thing at one time, keep changes small and viewable.
You can split it into multiple PRs, change vendor first.

**Comment by coocood** [comment]
@LiangShang 
This PR is un-mergeable, I'm sorry but I have to close it.
And I think `tidb-tools` is a better place for `tidb-ctl`, because it doesn't depend on any package in `tidb` repo.

**Pull request closed without merging**